From e321cbe7fae2d6a3d0dd5648c21e7dd19e92e6f3 Mon Sep 17 00:00:00 2001
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Fri, 30 Jan 2009 16:22:32 -0500
Subject: [PATCH] Procfs fix missing iter advance after lsm callout.

Other paths have the iter advancement, it looks like
someone simply missed this one.

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 fs/proc/base.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/fs/proc/base.c b/fs/proc/base.c
index ef9b9e2..9a8d11d 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -2733,8 +2733,10 @@ retry:
 			goto retry;
 		}
 		get_task_struct(iter.task);
-		if (security_task_lookup(iter.task))
+		if (security_task_lookup(iter.task)) {
+			iter.tgid += 1;
 			goto retry;
+		}
 	}
 	rcu_read_unlock();
 	return iter;
-- 
1.6.0.3

